From: Jiri Pirko <jpirko@redhat.com>
Date: Tue, 23 Feb 2010 23:17:07 +0000
Subject: [PATCH 091/207] net: convert multiple drivers to use
 netdev_for_each_mc_addr, part6

commit 567ec874d15b478c8eda7e9a5d2dcb05f13f1fb5 upstream.

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Restricted to drivers/net/tg3.{c,h}]
---
 drivers/net/tg3.c |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/drivers/net/tg3.c b/drivers/net/tg3.c
index 2d62017..9136c15 100644
--- a/drivers/net/tg3.c
+++ b/drivers/net/tg3.c
@@ -9459,15 +9459,12 @@ static void __tg3_set_rx_mode(struct net_device *dev)
 	} else {
 		/* Accept one or more multicast(s). */
 		struct dev_mc_list *mclist;
-		unsigned int i;
 		u32 mc_filter[4] = { 0, };
 		u32 regidx;
 		u32 bit;
 		u32 crc;
 
-		for (i = 0, mclist = dev->mc_list; mclist && i < netdev_mc_count(dev);
-		     i++, mclist = mclist->next) {
-
+		netdev_for_each_mc_addr(mclist, dev) {
 			crc = calc_crc (mclist->dmi_addr, ETH_ALEN);
 			bit = ~crc & 0x7f;
 			regidx = (bit & 0x60) >> 5;
-- 
1.7.5.4

